src/build.d

```diff
@@ -907,23 +893,51 @@ struct Dependency
     string[] targets; // list of all target files
     string[] sources; // list of all source files
     string[] rebuildSources; // Optional list of files that trigger a rebuild of this dependency
+    DependencyRef[] deps; // dependencies to build before this one
     string[] command; // the dependency command
     void delegate() commandFunction; // a custom dependency command which gets called instead of command
     string name; // name of the dependency that is e.g. written to the CLI when it's executed
     string[] trackSources;
+}
 
-    /**
-    Executes the dependency
-    */
-    auto run()
+class DependencyRef
```

Comment:
this should probably be a struct, no point having a reference to a ref, also `alias this` with classes are generally a bad idea (e.g. testing with `if (someRef)`).

Reply:
In this case it needs to be a reference type.  I could make it a strut and then change everything to `DependencyRef*`, but seems unnecessary since a class already does this.


But I agree that in most cases class are a bad idea.

Reply:
Plus the run call needs to be synchronized, and classes have that builtin.

Reply:
Ah I thought `dep` was a class too, class alias this to another class is where the problems lie, class alias this to a struct is fine. Carry on.